server/src/instant/reactive/ephemeral.clj

```diff
@@ -121,122 +127,45 @@
-                        (reduce (fn [store-v room-id]
-                                  (leave-room store-v app-id sess-id room-id))
-                                store-v
-                                rooms))]
-    (-> without-rooms
-        (dissoc-in [:sessions sess-id]))))
-
-(defn- set-presence
-  "Sets a session's presence data for a specific room."
-  [store-v app-id sess-id room-id data]
-  (assoc-in store-v [:rooms app-id room-id :data sess-id :data] data))
-
 ;; ---------
 ;; Hazelcast
 
-(defn handle-refresh-event [store-conn room-key room-id]
-  (let [room-data (.get (get-hz-rooms-map) room-key)
-        session-ids (filter (fn [sess-id]
-                              (rs/get-session @store-conn sess-id))
-                            (keys room-data))]
-    (doseq [sess-id session-ids
-            :let [q (:receive-q (rs/get-socket @store-conn sess-id))]
-            :when q]
-      (receive-queue/enqueue->receive-q q
-                                        {:op :refresh-presence
-                                         :app-id (:app-id room-key)
-                                         :room-id room-id
-                                         :data room-data
-                                         :session-id sess-id}))))
-
-(defn straight-jacket-refresh-event!
-  [store-conn {:keys [room-key room-id on-sent]}]
-  (try
-    (handle-refresh-event store-conn room-key room-id)
-    (catch Throwable t
-      (tracer/record-exception-span! t {:name "rooms-refresh-map/straight-jacket"}))
-    (finally (on-sent))))
-
-(defn start-refresh-map-worker [store-conn ch]
-  (loop [event (a/<!! ch)]
-    (if (nil? event)
-      (tracer/record-info! {:name "room-refresh-map/closed"})
-      (do
-        (straight-jacket-refresh-event! store-conn event)
-        (recur (a/<!! ch))))))
-
 (defn handle-event [store-conn ^DataAwareEntryEvent event]
-  (let [{:keys [app-id room-id] :as room-key} (.getKey event)]
-    (when (flags/use-hazelcast? app-id)
-      (when (seq (get-in @room-maps [:rooms room-key :session-ids]))
-        (let [room-data (.get (get-hz-rooms-map) room-key)]
-          (doseq [sess-id (keys room-data)
-                  :let [q (:receive-q (rs/get-socket @store-conn sess-id))]
-                  :when q]
-            (receive-queue/enqueue->receive-q q
-                                              {:op :refresh-presence
-                                               :app-id app-id
-                                               :room-id room-id
-                                               :data room-data
-                                               :session-id sess-id})))))))
+  (let [room-key                        (.getKey event)
+        {:keys [app-id room-id]}        room-key
+        {:keys [session-ids last-data]} (get-in @room-maps [:rooms room-key])]
+    (when (seq session-ids)
+      (let [room-data (.get (get-hz-rooms-map) room-key)
+            edits     (when last-data
```

Comment:
One rough thought I had: 

Is it necessary to derive edits? Perhaps we could broadcast `set-presence` events similar, to how we broadcast `server-broadcast` events. 

This would mean our resolution would be a bit less granular, but we wouldn't have to do a calculation to derive the edits. 

--- 

This could be something we layer on down the road though, as am open to either solution.